internal/services/storage/storage_table_resource_test.go

```diff
@@ -261,3 +276,41 @@ resource "azurerm_storage_table" "test" {
 }
 `, data.RandomInteger, data.Locations.Primary, data.RandomString, data.RandomInteger)
 }
+
+func (r StorageTableResource) resourceManagerId(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "acctestacc%s"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+
+  tags = {
+    environment = "staging"
+  }
+}
+
+resource "azurerm_storage_table" "test" {
+  name                 = "acctestst%d"
+  storage_account_name = azurerm_storage_account.test.name
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_role_assignment" "test" {
+  scope                = azurerm_storage_table.test.resource_manager_id
+  principal_id         = data.azurerm_client_config.current.object_id
+  role_definition_name = "Storage Table Data Contributor"
+}
```

Comment:
I included a `role_assignment` resource in this test configuration just to so Azure can verify the `resource_manager_id`, even though `role_assignment` is not the system under test.